fhir-cql/src/main/java/com/ibm/fhir/cql/engine/converter/impl/FhirTypeConverterImpl.java

```diff
@@ -0,0 +1,575 @@
+
+        if (value instanceof CqlType) {
+            result = true;
+        } else if (value instanceof java.math.BigDecimal || value instanceof java.lang.String || value instanceof java.lang.Integer
+                || value instanceof java.lang.Boolean) {
+            result = true;
+        } else {
+            result = false;
+        }
+
+        return result;
+    }
+    
+    @Override
+    public Object toFhirTuple(Tuple value) {
+        Object result = null;
+        if (value != null) {
+            throw new NotImplementedException("can't convert Tuples");
+        } 
+        return result;
+    }
+
+    @Override
+    public Object toCqlType(Object value) {
+        Object result = null;
+
+        if (isCqlType(value)) {
+            return value;
+        }
+
+        if (!isFhirType(value)) {
+            throw new IllegalArgumentException(String.format("can't convert %s to CQL type", value.getClass().getName()));
+        }
+
+        switch (value.getClass().getSimpleName()) {
+        // NOTE: There's no first class IdType in CQL, so the conversion to CQL Ids and
+        // back is asymmetric
+        case "Id":
+            result = toCqlId((Id) value);
+            break;
+        case "Boolean":
+            result = toCqlBoolean((com.ibm.fhir.model.type.Boolean) value);
+            break;
+        case "Integer":
+            result = toCqlInteger((com.ibm.fhir.model.type.Integer) value);
+            break;
+        case "Decimal":
+            result = toCqlDecimal((com.ibm.fhir.model.type.Decimal) value);
+            break;
+        case "Date":
+            result = toCqlDate((com.ibm.fhir.model.type.Date) value);
+            break;
+        // NOTE: There's no first class InstantType in CQL, so the conversation to CQL
+        // DateTime and back is asymmetric
+        case "Instant":
+        case "DateTime":
+            result = toCqlDateTime((com.ibm.fhir.model.type.DateTime) value);
+            break;
+        case "Time":
+            result = toCqlTime((com.ibm.fhir.model.type.Time) value);
+            break;
+        case "String":
+            result = toCqlString((com.ibm.fhir.model.type.String) value);
+            break;
+        case "Quantity":
+            result = toCqlQuantity((com.ibm.fhir.model.type.Quantity) value);
+            break;
+        case "Ratio":
+            result = toCqlRatio((com.ibm.fhir.model.type.Ratio) value);
+            break;
+        // case "AnyType": result = toCqlAny((IBase)value);break;
+        case "Coding":
+            result = toCqlCode((com.ibm.fhir.model.type.Coding) value);
+            break;
+        case "CodeableConcept":
+            result = toCqlConcept((com.ibm.fhir.model.type.CodeableConcept) value);
+            break;
+        case "Period":
+        case "Range":
+            result = toCqlInterval((com.ibm.fhir.model.type.Element) value);
+            break;
+        // case "Tuple": return toCqlTuple((IBase) value);

```

Comment:
same thing through out... are these intentionally here? if so, best to add a comment